app/controllers/work_packages/bulk_controller.rb

```diff
@@ -39,35 +39,19 @@ class WorkPackages::BulkController < ApplicationController
   include IssuesHelper
 
   def edit
-    @work_packages.sort!
-    @available_statuses = @projects.map { |p|Workflow.available_statuses(p) }.inject { |memo, w|memo & w }
-    @custom_fields = @projects.map(&:all_work_package_custom_fields).inject { |memo, c|memo & c }
-    @assignables = @projects.map(&:possible_assignees).inject { |memo, a| memo & a }
-    @responsibles = @projects.map(&:possible_responsibles).inject { |memo, a| memo & a }
-    @types = @projects.map(&:types).inject { |memo, t| memo & t }
+    WorkPackageBulkUpdateService.new(@work_packages, @projects).run(params)
+    .available_attributes.each do |key, value|
+      self.instance_variable_set(:"@#{key}", value)
+    end
   end
 
   def update
-    @work_packages.sort!
-
-    unsaved_work_package_ids = []
-
-    @work_packages.each do |work_package|
-      work_package.reload
-      work_package.add_journal(User.current, params[:notes])
-
-      # filter parameters by whitelist and add defaults
-      attributes = parse_params_for_bulk_work_package_attributes params, work_package.project
-      work_package.assign_attributes attributes
-
-      call_hook(:controller_work_package_bulk_before_save,  params: params, work_package: work_package)
-      JournalObserver.instance.send_notification = params[:send_notification] == '0' ? false : true
-      unless work_package.save
-        unsaved_work_package_ids << work_package.id
-      end
+    WorkPackageBulkUpdateService.new(@work_packages, @projects).save(params).each do |key, value|
+      self.instance_variable_set(:"@#{key}", value)
     end
-    set_flash_from_bulk_save(@work_packages, unsaved_work_package_ids)
-    redirect_back_or_default({ controller: '/work_packages', action: :index, project_id: @project }, false)
+
+    set_flash_from_bulk_save(@work_packages, @unsaved_work_package_ids)
+    redirect_back_or_default({controller: '/work_packages', action: :index, project_id: @project}, false)
```

Comment:
Line is too long. [105/99]<br>Space inside { missing.<br>Space inside } missing.